src/main/java/org/fogbowcloud/manager/core/models/orders/ComputeOrder.java

```diff
@@ -1,92 +1,73 @@
 package org.fogbowcloud.manager.core.models.orders;
 
-import org.fogbowcloud.manager.core.models.orders.instances.OrderInstance;
 import org.fogbowcloud.manager.core.models.token.Token;
 
-import javax.persistence.*;
+import java.util.UUID;
 
-@Entity
-@DiscriminatorValue(value = "compute")
 public class ComputeOrder extends Order {
 
-	@Column(name = "vCPU")
 	private int vCPU;
 
 	/** Memory attribute, must be set in MB. */
-	@Column(name = "memory")
 	private int memory;
 
 	/** Disk attribute, must be set in GB. */
-	@Column(name = "disk")
 	private int disk;
-	
-	@Column(name = "image_name")
+
 	private String imageName;
 
-	@OneToOne(cascade = CascadeType.ALL)
-	@JoinColumn(name = "user_data_id")
 	private UserData userData;
 
-	public ComputeOrder(){ }
-
-	public ComputeOrder(OrderState orderState, Token localToken, Token federationToken, String requestingMember,
-						String providingMember, OrderInstance orderInstance, long fulfilledTime, int vCPU, int memory,
-						int disk, String imageName, UserData userData) {
-		super(orderState, localToken, federationToken, requestingMember, providingMember, orderInstance, fulfilledTime);
+	/**
+	 * Creating Order with predefined Id.
+	 */
+	public ComputeOrder(String id, Token localToken, Token federationToken, String requestingMember,
+			String providingMember, int vCPU, int memory, int disk, String imageName, UserData userData) {
+		super(id, localToken, federationToken, requestingMember, providingMember);
 		this.vCPU = vCPU;
 		this.memory = memory;
 		this.disk = disk;
-		this.imageName= imageName;
+		this.imageName = imageName;
 		this.userData = userData;
 	}
 
-	public int getvCPU() {
-		return vCPU;
+	public ComputeOrder(Token localToken, Token federationToken, String requestingMember, String providingMember,
+			int vCPU, int memory, int disk, String imageName, UserData userData) {
+		this(UUID.randomUUID().toString(), localToken, federationToken, requestingMember, providingMember, vCPU, memory,
+				disk, imageName, userData);
 	}
 
-	public void setvCPU(int vCPU) {
-		this.vCPU = vCPU;
+	public int getvCPU() {
+		return vCPU;
 	}
 
 	public int getMemory() {
 		return memory;
 	}
 
-	public void setMemory(int memory) {
-		this.memory = memory;
-	}
-
 	public int getDisk() {
 		return disk;
 	}
 
-	public void setDisk(int disk) {
-		this.disk = disk;
-	}
-	
 	public String getImageName() {
 		return imageName;
 	}
 
-	public void setImageName(String imageName) {
-		this.imageName = imageName;
-	}
-
 	public UserData getUserData() {
 		return userData;
 	}
 
-	public void setUserData(UserData userData) {
-		this.userData = userData;
-	}
-	
 	@Override
 	public OrderType getType() {
 		return OrderType.COMPUTE;
 	}
 
+	/**
```

Comment:
Don't we agree on removing these handle, process* methods from Order classes?

Reply:
No, we agree in take the instance provider from the Order class and put it at OpenProcessor Thread. The handleOpenOrder is necessary to prepare an Order to change your state from Open to Spawning, as mentioned at the method javadoc.

Reply:
ok

Reply:
Done, [here](https://github.com/fogbow/fogbow-manager-core/pull/13/commits/558d5c720295ef89068ce1944289e496a2a61972).